File: python/with_java/programmers/level 1/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    for num in range(1, number+1):
        cnt = 0
        for n in range(1, int(num ** 0.5)+1):
            if num % n != 0:
                continue
            if num // n != n:
                cnt += 1
            cnt += 1
        if cnt > limit:
            cnt = power
        answer += cnt
        
    return answer

File: java/src/main/java/ps_traning/programmers/level_1/기사단원의_무기.java
package ps_traning.programmers.level_1;

public class 기사단원의_무기 {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for (int num = 1; num <= number; num++) {
            int cnt = 0;
            for (int n = 1; n <= Math.sqrt(num); n++) {
                if (num % n != 0) continue;
                if (num / n != n) cnt++;
                cnt++;
            }
            if (cnt > limit) {
                cnt = power;
            }
            answer += cnt;
        }
        return answer;
    }
}
